src/evaluators/CallExpression.js

```diff
@@ -112,7 +112,7 @@ function generateRuntimeCall(
   if (propName !== undefined && typeof propName !== "string") args.push(propName);
   args = args.concat(ArgumentListEvaluation(realm, strictCode, env, ast.arguments));
   for (let arg of args) {
-    if (arg !== func) {
+    if (arg !== func && !func.safeFromLeaks) {
```

Comment:
This is only checking the top value not if it is nested.

Reply:
I don't understand the motivation behind the original condition (why excludes passing the func in any argument position from leaking?), or the amended one (should it be `!arg.safeFromLeaks`? if not, why not?)

Reply:
Also, there are many calls to `.leakValue` in the code. Why is `.safeFromLeaks` only respected here?

Reply:
I’m not trying to check against objects, but rather than root visit to the matching abstract function. I think the whole naming of `leak` is complicating things somewhat here too. We should rename to `escape`. This bit of logic makes more sense then - “this abstract function is safe when evaluating its contents for escape analysis”.